eng/notSupported.SourceBuild.targets

```diff
@@ -0,0 +1,25 @@
+<Project InitialTargets="_RedefineNotSupportedSourceFile">
+
+  <Target Name="_RedefineNotSupportedSourceFile"
+          Condition="'$(DotNetBuildFromSource)' == 'true' and
+                     ('$(GeneratePlatformNotSupportedAssembly)' == 'true' or '$(GeneratePlatformNotSupportedAssemblyMessage)' != '')">
+                     
+    <Error Condition="'$(DotNetSourceBuildIntermediatePath)' == ''" 
+           Text="'DotNetSourceBuildIntermediatePath' must be specified when 'DotNetBuildFromSource' is true" />
+           
+    <PropertyGroup>
+      <_notSupportedSourceDirectory>$([MSBuild]::NormalizeDirectory('$(DotNetSourceBuildIntermediatePath)', '$(MSBuildProjectName)', '$(TargetGroup)-$(OSGroup)'))</_notSupportedSourceDirectory>
+      <NotSupportedSourceFile>$(_notSupportedSourceDirectory)$(TargetName).notsupported.cs</NotSupportedSourceFile>
```

Comment:
@ericstj who creates that file?

Reply:
GenAPI.  https://github.com/dotnet/arcade/blob/a1805103791e43031355e11c0d037bca803a9593/src/Microsoft.DotNet.GenAPI/build/Microsoft.DotNet.GenAPI.targets#L90